### ISSUE TITLE
Flycheck does not show non-exhaustive pattern match warnings.

### ISSUE THREAD

**Issue by ntc2** [flycheck/flycheck#722]
Consider for example this program:
```
{-# OPTIONS_GHC -fwarn-incomplete-patterns #-}
module Test where

f :: Maybe t -> t
f (Just x) = x
```

Compiling it with no options shows a non-exhaustive pattern match warning:
```
$ ghc Test.hs
[1 of 1] Compiling Test             ( Test.hs, Test.o )

Test.hs:5:1: Warning:
    Pattern match(es) are non-exhaustive
    In an equation for ‘f’: Patterns not matched: Nothing
```

But in `flycheck-mode` using `haskell-ghc` checker, no warning is shown.

The problem is that Flycheck passes `-fno-code` to GHC (https://github.com/flycheck/flycheck/blob/2d6816c6aad9a0d0cbb2d82a77dc9dfa6261a89d/flycheck.el#L6441). It is a known GHC bug that `-fno-code` disables non-exhaustive pattern warnings (https://ghc.haskell.org/trac/ghc/ticket/10600).

So, this is not really Flycheck's fault, and perhaps @lunaryorn will close this issue and wait for the upstream fix in GHC 7.12.  I'm creating this issue to document the problem and it's cause, in case anyone else runs into it.

**Comment by lunaryorn** [comment]
@ntc2 See #708, we'll drop `-fno-code` at some point.  Closing.
